clients/bigquery/bigquery.go

```diff
@@ -35,10 +36,54 @@ type Store struct {
 	db.Store
 }
 
-func (s *Store) getTableConfig(tableData *optimization.TableData) (*types.DwhTableConfig, error) {
-	return utils.GetTableConfig(utils.GetTableCfgArgs{
+func (s *Store) PrepareTemporaryTable(tableData *optimization.TableData, tableConfig *types.DwhTableConfig, tempTableName string, _ types.AdditionalSettings) error {
+	// 1. Create Temporary table
+	tempAlterTableArgs := ddl.AlterTableArgs{
+		Dwh:               s,
+		Tc:                tableConfig,
+		FqTableName:       fmt.Sprintf("%s_%s", s.ToFullyQualifiedName(tableData, false), tableData.TempTableSuffix()),
+		CreateTable:       true,
+		TemporaryTable:    true,
+		ColumnOp:          constants.Add,
+		UppercaseEscNames: &s.config.SharedDestinationConfig.UppercaseEscapedNames,
+	}
+
+	if err := ddl.AlterTable(tempAlterTableArgs, tableData.ReadOnlyInMemoryCols().GetColumns()...); err != nil {
+		return fmt.Errorf("failed to create temp table: %w", err)
+	}
+
+	// 2. Cast all the data into rows.
+	var rows []*Row
+	additionalDateFmts := s.config.SharedTransferConfig.TypingSettings.AdditionalDateFormats
+	for _, value := range tableData.Rows() {
+		data := make(map[string]bigquery.Value)
+		for _, col := range tableData.ReadOnlyInMemoryCols().GetColumnsToUpdate(s.config.SharedDestinationConfig.UppercaseEscapedNames, nil) {
+			colKind, _ := tableData.ReadOnlyInMemoryCols().GetColumn(col)
+			colVal, err := castColVal(value[col], colKind, additionalDateFmts)
+			if err != nil {
+				return fmt.Errorf("failed to cast col %s: %w", col, err)
+			}
+
+			if colVal != nil {
+				data[col] = colVal
+			}
+		}
+
+		rows = append(rows, NewRow(data))
+	}
+
+	// 3. Load the data into the temporary table.
+	return s.putTable(context.Background(), tableData.TopicConfig.Database, tempTableName, rows)
```

Comment:
Was it a bug that we were inserting data into `fmt.Sprintf("%s_%s", tableData.RawName(), tableData.TempTableSuffix())` before?

Reply:
No

Reply:
```go
fmt.Sprintf("%s_%s", tableData.RawName(), tableData.TempTableSuffix())
```
This will output the table name along with the suffix.

Reply:
`ToFullyQualifiedName(...)` will output the absolute table name